File: src/Web/Controllers/MessageController.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace AspNetCore
{
	[Route("messages")]
	public class MessagesController : Controller
	{
		private readonly MessageDbContext context;

		public MessagesController(MessageDbContext context)
		{
			this.context = context;
		}

		[HttpGet("")]
		public async Task<IActionResult> GetAllMessages()
		{
			var messages = await context.Messages
				.Select(message => new MessageViewModel {Id = message.Id, Text = message.Text})
				.ToArrayAsync();

			var viewModel = new MessagesViewModel
			{
				Messages = messages
			};

			return Json(viewModel);
		}

		[HttpPost("add")]
		public async Task<IActionResult> AddMessage([FromBody] MessageViewModel model)
		{
			var message = new Message
			{
				Text = model.Text
			};
			
			context.Messages.Add(message);

			await context.SaveChangesAsync();

			return Ok();
		}
	}
}

File: src/Web/React/src/Components/Message/MessageController.ts
import axios, { AxiosPromise } from 'axios'
import { MessageViewModel, MessagesViewModel } from 'components/Message'

export interface MessageController {
	getAllMessages(): AxiosPromise<MessagesViewModel>;

	addMessage(message: MessageViewModel): AxiosPromise;
}

class MessageControllerClass implements MessageController {

	public getAllMessages(): AxiosPromise<MessagesViewModel> {
		return axios.get<MessagesViewModel>('/messages')
	}

	public addMessage(message: MessageViewModel): AxiosPromise {
		return axios.post('/messages/add', message)
	}
}

export const Controller: MessageController = new MessageControllerClass()